src/System.Linq/tests/MinMaxTests.cs

```diff
@@ -0,0 +1,373 @@
+            var minusTen = new[] { -100F, -15, -50, -10 };
+            var thousand = new[] { -16F, 0, 50, 100, 1000 };
+            Assert.Equal(10F, ten.Max());
+            Assert.Equal(-10F, minusTen.Max());
+            Assert.Equal(1000F, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<float>().Max());
+        }
+
+        [Fact]
+        public void MinDouble()
+        {
+            var one = Enumerable.Range(1, 10).Select(i => (double)i).ToArray();
+            var minusTen = new[] { -1D, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000D, 100, 200, 1000 };
+            Assert.Equal(1D, one.Min());
+            Assert.Equal(-10D, minusTen.Min());
+            Assert.Equal(100D, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<double>().Min());
+        }
+
+        [Fact]
+        public void MaxDouble()
+        {
+            var ten = Enumerable.Range(1, 10).Select(i => (double)i).ToArray();
+            var minusTen = new[] { -100D, -15, -50, -10 };
+            var thousand = new[] { -16D, 0, 50, 100, 1000 };
+            Assert.Equal(10D, ten.Max());
+            Assert.Equal(-10D, minusTen.Max());
+            Assert.Equal(1000D, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<double>().Max());
+        }
+
+        [Fact]
+        public void MinDecimal()
+        {
+            var one = Enumerable.Range(1, 10).Select(i => (decimal)i).ToArray();
+            var minusTen = new[] { -1M, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000M, 100, 200, 1000 };
+            Assert.Equal(1M, one.Min());
+            Assert.Equal(-10M, minusTen.Min());
+            Assert.Equal(100M, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<decimal>().Min());
+        }
+
+        [Fact]
+        public void MaxDecimal()
+        {
+            var ten = Enumerable.Range(1, 10).Select(i => (decimal)i).ToArray();
+            var minusTen = new[] { -100M, -15, -50, -10 };
+            var thousand = new[] { -16M, 0, 50, 100, 1000 };
+            Assert.Equal(10M, ten.Max());
+            Assert.Equal(-10M, minusTen.Max());
+            Assert.Equal(1000M, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<decimal>().Max());
+        }
+
+        [Fact]
+        public void MinNullableInt()
+        {
+            var one = Enumerable.Range(1, 10).Select(i => (int?)i).ToArray();
+            var minusTen = new[] { default(int?), -1, -10, 10, 200, 1000 };
+            var hundred = new[] { default(int?), 3000, 100, 200, 1000 };
+            Assert.Equal(1, one.Min());
+            Assert.Equal(-10, minusTen.Min());
+            Assert.Equal(100, hundred.Min());
+            Assert.Equal(null, Enumerable.Empty<int?>().Min());
+            Assert.Equal(null, Enumerable.Repeat(default(int?), 100).Min());
```

Comment:
It'd be good to have cases where there are null and non-null values interspersed throughout, not just where the first element is null or where all elements are null.  Plus the same comments from earlier, e.g. testing a single-element array.